src/Features/Core/Portable/InlineMethod/AbstractInlineMethodRefactoringProvider.InlineContext.cs

```diff
@@ -0,0 +1,558 @@
+                InlineSyntaxNode = inlineSyntaxNode;
+                SyntaxNodeToReplace = syntaxNodeToReplace;
+                ContainsAwaitExpression = containsAwaitExpression;
+            }
+
+            public static async Task<InlineMethodContext> GetInlineContextAsync(
+                AbstractInlineMethodRefactoringProvider<TInvocationSyntaxNode, TExpressionSyntax, TArgumentSyntax> inlineMethodRefactoringProvider,
+                ISyntaxFacts syntaxFacts,
+                IPrecedenceService precedenceService,
+                Document document,
+                SemanticModel semanticModel,
+                SyntaxNode calleeInvocationSyntaxNode,
+                IMethodSymbol calleeMethodSymbol,
+                SyntaxNode calleeMethodDeclarationSyntaxNode,
+                MethodParametersInfo methodParametersInfo,
+                SyntaxNode root,
+                CancellationToken cancellationToken)
+            {
+                var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+                var rawInlineSyntaxNode = inlineMethodRefactoringProvider.GetInlineStatement(calleeMethodDeclarationSyntaxNode);
+                var inlineSyntaxNode = rawInlineSyntaxNode;
+                // The replacement/insertion work done might cause naming conflict.
+                // So first compute a mapping from symbol to newName (renameTable)
+                // Cases are:
+                // 1. Replace the callee's parameter withe caller's identifier might cause conflict in callee.
+                // 2. Use the parameter's name to generate declarations in caller might cause conflict in the caller
+                // In either case, rename the symbol in Callee.
+                var parametersWithIdentifierArgumentToName = methodParametersInfo.ParametersWithIdentifierArgument
+                    .Select(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol,
+                            semanticModel.GetSymbolInfo(parameterAndArgument.identifierSyntaxNode, cancellationToken)
+                                .GetAnySymbol()?.Name))
+                    .Where(parameterAndName => parameterAndName.Name != null)
+                    .ToImmutableArray();
+
+                var parametersWithVariableDeclarationArgumentToName = methodParametersInfo.ParametersWithVariableDeclarationArgument
+                    .Select(parameterAndArgument =>
+                        (parameterAndArgument.parameterSymbol,
+                            semanticModel
+                                .GetSymbolInfo(parameterAndArgument.variableDeclarationNode, cancellationToken)
+                                .GetAnySymbol()?.Name))
+                    .Where(parameterAndName => parameterAndName.Name != null)
+                    .ToImmutableArray();
+
+                var allSymbolsOfCaller = semanticModel.LookupSymbols(calleeInvocationSyntaxNode.Span.End);
```

Comment:
so i think you're reimplementing logic we have elsewhere.  this is just trying to pick names taht don't conflict right?  You can use ISemanticFactsService.GenerateUniqueLocalName for this sort of thing.

Reply:
Ah, that will simplify this code a lot 😃